fdbclient/include/fdbclient/MetaclusterManagement.actor.h

```diff
@@ -1152,31 +1153,46 @@ struct CreateTenantImpl {
 
 			if (groupEntry.present()) {
 				ASSERT(groupEntry.get().assignedCluster.present());
+				if (self->preferAssignedCluster) {
+					ASSERT(groupEntry.get().assignedCluster.get() == self->tenantEntry.assignedCluster.get());
```

Comment:
This shouldn't be an assert since a user could errantly select an invalid cluster. I suppose here we could also use `invalid_tenant_configuration` or introduce a new error.